mmrotate/datasets/dota.py

```diff
@@ -1,361 +1,142 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 import glob
-import os
 import os.path as osp
-import re
-import tempfile
-import time
-import zipfile
-from collections import defaultdict
-from functools import partial
+from typing import List, Tuple
 
-import mmcv
-import numpy as np
-import torch
-from mmcv.ops import nms_rotated
-from mmdet.datasets.custom import CustomDataset
+from mmengine.dataset import BaseDataset
 
-from mmrotate.core import eval_rbbox_map, obb2poly_np, poly2obb_np
-from .builder import ROTATED_DATASETS
+from mmrotate.registry import DATASETS
 
 
-@ROTATED_DATASETS.register_module()
-class DOTADataset(CustomDataset):
+@DATASETS.register_module()
+class DOTADataset(BaseDataset):
     """DOTA dataset for detection.
 
+    Note: ``ann_file`` in DOTADataset is different from the BaseDataset.
+    In BaseDataset, it is the path of an annotation file. In DOTADataset,
+    it is the path of a folder containing XML files.
+
     Args:
-        ann_file (str): Annotation file path.
-        pipeline (list[dict]): Processing pipeline.
-        version (str, optional): Angle representations. Defaults to 'oc'.
-        difficulty (bool, optional): The difficulty threshold of GT.
+        img_shape (tuple[int]): The shape of images. Due to the huge size
+            of the remote sensing image, we will cut it into slices with
+            the same shape. Defaults to (1024, 1024).
+        diff_thr (int): The difficulty threshold of ground truth. Bboxes
+        with difficulty higher than it will be ignored. The range of this
+        value should be non-negative integer. Defaults to 100.
```

Comment:
```suggestion
        diff_thr (int): The difficulty threshold of ground truth. Bboxes
            with difficulty higher than it will be ignored. The range of this
            value should be non-negative integer. Defaults to 100.
```